docs/docs/exit-codes.md

```diff
@@ -8,16 +8,31 @@ Marathon follows the [Let-it-crash](https://www.reactivedesignpatterns.com/patte
-|-----------|----------------------------------------------------------------------|
-|100        | `ZookeeperConnectionFailure` - Could not connect to Zookeeper        |
-|101        | `ZookeeperConnectionLost` - Lost connect to Zookeeper                |
-|102        | `PluginInitializationFailure` - Could not load plugin                |
-|103        | `LeadershipLoss` - Lost leadership                                   |
-|104        | `LeadershipEndedFaulty` - Leadership ended with an error             |
-|105        | `LeadershipEndedGracefully` - Leadership ended without an error      |
-|106        | `MesosSchedulerError` - The Mesos scheduler driver had an error      |
-|107        | `UncaughtException` - An internal unknown error could not be handled |
-|108        | The Framework ID could not be read.                                  |
-|109        | Provided LibMesos version is incompatible.                           |
-|137        | Killed by an external process or uncaught exception                  |
+| Exit Code | Reason ......                                                                 |
+|-----------|-------------------------------------------------------------------------------|
+|100        | `ZookeeperConnectionFailure` - Could not connect to Zookeeper                 |
+|101        | `ZookeeperConnectionLost` - Lost connect to Zookeeper                         |
+|102        | `PluginInitializationFailure` - Could not load plugin                         |
+|103        | `LeadershipLoss` - Lost leadership                                            |
+|104        | `LeadershipEndedFaulty` - Leadership ended with an error                      |
+|105        | `LeadershipEndedGracefully` - Leadership ended without an error               |
+|106        | `MesosSchedulerError` - The Mesos scheduler driver had an error               |
+|107        | `UncaughtException` - An internal unknown error could not be handled          |
+|108        | `FrameworkIdMissing` The Framework ID could not be read                       |
+|109        | `IncompatibleLibMesos` Provided LibMesos version is incompatible              |
+|110        | `FrameworkHasBeenRemoved` Framework has been removed via Mesos teardown call  |
+|137        | Killed by an external process or uncaught exception                           |
+
+## Troubleshooting Exit Codes
+
+### 102 Plugin Initialization Failure
+
+One or more configured plugins could not be initialized. Please check your plugin configuration and the compatibility of the loaded plugin.
+
+### 108 Missing Framework ID
+
+The configured zookeeper path contains references to launched app or pod instances. However, it does not provide a Framework ID for registration with Mesos. Marathon will not register without a Framework ID, since this would orphan any existing app or pod instances. Please see the documentation on [Framework ID registration](framework-id.html).
```

Comment:
```suggestion
The configured zookeeper path contains references to the launched app or pod instances but it does not contain a Framework ID for registration with Mesos. Marathon will not register without a Framework ID since this would orphan any existing app or pod instances. Please see the documentation on [Framework ID registration](framework-id.html).
```